### PR TITLE
Preferred shell

### LINKED ISSUE #888: Preferred shell options
Users should be able to select their preferred shell in settings. In cases where no selection is made, we should use the OS default.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #888

### CODE REVIEW

File: app/src/lib/dispatcher/app-store.ts [hunk 1]
```diff
@@ -864,13 +875,26 @@ export class AppStore {
...
     this.accountsStore.refresh()
   }
 
+  /** Update the menu with the names of the user's preferred apps. */
+  private updatePreferredAppMenuItemLabels() {
+    updatePreferredAppMenuItemLabels({
+      editor: `Open in ${this.selectedExternalEditor}`,
+      shell: __DARWIN__
+        ? `Open in ${this.selectedShell}`
+        : 'Op&en command prompt',
```
Comment (joshaber) [inline app/src/lib/dispatcher/app-store.ts]: Do we wanna use the same `Open in X` wording as Mac?

File: app/src/lib/shells/win32.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  return Default
+}
+
+export async function getAvailableShells(): Promise<ReadonlyArray<Shell>> {
+  const shells = [Shell.Cmd]
+
+  const powerShell = await readRegistryKeySafe(
+    'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\PowerShell\\3\\PowerShellEngine'
```
Comment (joshaber) [inline app/src/lib/shells/win32.ts]: I'm not sure if this is the appropriate key...

File: app/src/main-process/menu/build-default-menu.ts [hunk 3]
```diff
@@ -3,15 +3,20 @@ import { ensureItemIds } from './ensure-item-ids'
...
-import { ExternalEditor } from '../../models/editors'
 
 import { log } from '../log'
 
+const defaultEditorLabel = __DARWIN__
+  ? 'Open in External Editor'
+  : 'Open in external editor'
+const defaultShellLabel = __DARWIN__
+  ? 'Open in Terminal'
+  : 'Op&en command prompt'
```
Comment (joshaber) [inline app/src/main-process/menu/build-default-menu.ts]: Same question as above whether wanna use `Open in X`.

File: app/src/ui/repositories-list/repository-list-item.tsx [hunk 4]
```diff
@@ -87,7 +90,9 @@ export class RepositoryListItem extends React.Component<
 
     const items: ReadonlyArray<IMenuItem> = [
       {
-        label: __DARWIN__ ? 'Open in Terminal' : 'Open command prompt',
+        label: __DARWIN__
+          ? `Open in ${this.props.shellLabel}`
+          : 'Open command prompt',
```
Comment (joshaber) [inline app/src/ui/repositories-list/repository-list-item.tsx]: And same question here about `Open in X`.

**Comment by joshaber** [comment]
This is ready 🙏 

I definitely bumbled my way through the Windows side, mostly cargo-culting, so I'm sure that could use some more careful 👀

File: app/src/lib/dispatcher/app-store.ts (hunk 1 above)
Comment (shiftkey) [inline app/src/lib/dispatcher/app-store.ts]: Yessir

File: app/src/lib/shells/win32.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { spawn } from 'child_process'
+import * as Path from 'path'
+import { assertNever } from '../fatal-error'
+import { readRegistryKeySafe } from '../registry'
+import { IFoundShell } from './found-shell'
+
+export enum Shell {
+  Cmd = 'cmd',
```
Comment (joshaber) [inline app/src/lib/shells/win32.ts]: Is this the right name? `cmd.exe`? `Command prompt`?

Comment (shiftkey) [inline app/src/lib/shells/win32.ts]: `cmd.exe` is the executable, but Command Prompt is the friendly name. Here's how it looks in the start menu when I search for it:

**Comment by michaelpdebus5** [comment]
Is this in my file? Wow. I do not know how to fix the mess that is around past devices and current ones. I do see things from 2003-2013 that was impossible to be me...

**Comment by joshaber** [comment]
Feel free to carry this forward while I'm out, or I can pick it back up when I'm back.

**Pull request merged**